core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -410,53 +443,58 @@
+                      valid-from (.readLong valid-from-rdr idx)
+                      valid-to (.readLong valid-to-rdr idx)]
+                  (when (> valid-from valid-to)
+                    (throw (err/runtime-err :xtdb.indexer/invalid-valid-times
+                                            {:valid-from (util/micros->instant valid-from)
+                                             :valid-to (util/micros->instant valid-to)})))
+
+                  (.writeRowId live-table new-row-id)
+
+                  (doseq [^IRowCopier update-col-copier (vals update-col-copiers)]
+                    (.copyRow update-col-copier idx))
+
+                  (if-let [live-row (.liveRow live-table old-row-id)]
+                    (doseq [^IIndirectVector live-col live-row
+                            :when (not (contains? update-col-copiers (.getName live-col)))]
+                      (doto ^IRowCopier (->live-row-copier live-col)
+                        (.copyRow 0)))
+
+                    (-> (when-let [{:keys [^long chunk-idx ^long block-idx, col-names]}
+                                   (meta/row-id->chunk metadata-mgr table old-row-id)]
+                          (let [idx @(-> (.getBuffer buffer-pool (meta/->chunk-obj-key chunk-idx table "_row_id"))
+                                         (util/then-apply #(row-id->idx % block-idx old-row-id)))]
+
+                            (doseq [^String col-name col-names
+                                    :when (not (contains? update-col-copiers col-name))]
+                              @(-> (.getBuffer buffer-pool (meta/->chunk-obj-key chunk-idx table col-name))
+                                   (util/then-apply
+                                     (fn [buf]
+                                       (with-open [chunks (util/->chunks buf {:block-idxs (doto (RoaringBitmap.)
+                                                                                            (.add block-idx))
+                                                                              :close-buffer? true})]
+                                         (-> chunks
+                                             (.forEachRemaining (reify Consumer
+                                                                  (accept [_ block-root]
+                                                                    (let [col (iv/->direct-vec (.getVector ^VectorSchemaRoot block-root col-name))]
+                                                                      (doto ^IRowCopier (->live-row-copier col)
+                                                                        (.copyRow idx))))))))))))))
+                        (util/rethrowing-cause)))
+
+                  (.logPut log-op-idxer iid new-row-id valid-from valid-to)
+                  ;; TODO scan does not currently pass through eid, so not possible to do eid -> iid
+                  (.logPut temporal-log-op-idxer (long->byte-hash iid) new-row-id valid-from valid-to)
```

Comment:
The `xt$id` column doesn't come through here with the in-rel. So I can't do `(->iid eid)`. Should that be changed?

Reply:
Tracked in #2611.